python/setup.py

```diff
@@ -227,7 +227,9 @@ def get_packages(self):
 if setup_spec.type == SetupType.RAY:
     pandas_dep = "pandas >= 1.3"
     numpy_dep = "numpy >= 1.20"
-    pyarrow_dep = "pyarrow >= 6.0.1,<18"
+    pyarrow_dep = (
+        "pyarrow >= 6.0.1,<18" if sys.platform == "darwin" else "pyarrow >= 6.0.1"
+    )
```

Comment:
this should be in the python deps constraint format (not in python conditions).

you can use line 235 as an example: `"cupy-cuda12x; sys_platform != 'darwin'"`